pkg/liqoctl/install/eks/provider.go

```diff
@@ -0,0 +1,386 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/iam"
+	flag "github.com/spf13/pflag"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"github.com/liqotech/liqo/pkg/liqoctl/install/provider"
+)
+
+const (
+	providerPrefix = "eks"
+)
+
+type eksProvider struct {
+	region      string
+	clusterName string
+
+	endpoint    string
+	serviceCIDR string
+	podCIDR     string
+
+	iamLiqoUser iamLiqoUser
+}
+
+type iamLiqoUser struct {
+	userName   string
+	policyName string
+
+	accessKeyID     string
+	secretAccessKey string
+}
+
+// NewProvider initializes a new EKS provider struct.
+func NewProvider() provider.InstallProviderInterface {
+	return &eksProvider{}
+}
+
+// ValidateCommandArguments validates specific arguments passed to the install command.
+func (k *eksProvider) ValidateCommandArguments(flags *flag.FlagSet) (err error) {
+	k.region, err = flags.GetString(prefixedName("region"))
+	if err != nil {
+		return err
+	}
+	if k.region == "" {
+		err := fmt.Errorf("--eks.region not provided")
+		return err
+	}
```

Comment:
It would be nice, but the problem is that they are required only if the provider is set to `eks`

To make it possible to use the required check we have to move each provider in a different subcommand, for example:
```bash
liqoctl install eks ... [flags]
```
instead of
```bash
liqoctl install --provider eks ... [flags]
```

Reply:
Got it. Personally, I would go that way in the long term, as it also allows to better divide the flags.